src/hooks/protections/received_event.cpp

```diff
@@ -582,23 +582,26 @@ namespace big
 
 			if (g_local_player && g_local_player->m_net_object && g_local_player->m_net_object->m_object_id == net_id)
 			{
+				weapon_item weapon = g_gta_data_service->weapon_by_hash(hash);
 				g_notification_service->push_warning("PROTECTIONS"_T.data(),
-				    std::vformat("REMOVE_WEAPON_ATTEMPT"_T, std::make_format_args(source_player->get_name())));
+					std::format("{} {} {}.", source_player->get_name(), "REMOVE_WEAPON_ATTEMPT"_T, weapon.m_display_name));
 				g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
 				return;
 			}
 
 			buffer->Seek(0);
 			break;
 		}
-		case eNetworkEvents::REMOVE_ALL_WEAPONS_EVENT:
+		case eNetworkEvents::GIVE_WEAPON_EVENT:
 		{
-			int net_id = buffer->Read<int>(13);
+			std::int16_t net_id = buffer->Read<std::int16_t>(13);
+			Hash hash           = buffer->Read<Hash>(32);
 
 			if (g_local_player && g_local_player->m_net_object && g_local_player->m_net_object->m_object_id == net_id)
 			{
+				weapon_item weapon = g_gta_data_service->weapon_by_hash(hash);
 				g_notification_service->push_warning("PROTECTIONS"_T.data(),
-				    std::vformat("REMOVE_ALL_WEAPONS_ATTEMPT"_T, std::make_format_args(source_player->get_name())));
+				    std::format("{} {} {}.", source_player->get_name(), "GIVE_WEAPON_ATTEMPT"_T, weapon.m_display_name));

```

Comment:
This change will fuck over every other language that had this key translated.

Reply:
No it won't, this key is new.